diff --git a/Source/WebCore/bindings/js/JSDOMWrapper.h b/Source/WebCore/bindings/js/JSDOMWrapper.h
index 91cd11aaf6..86ee2a1aba 100644
--- a/Source/WebCore/bindings/js/JSDOMWrapper.h
+++ b/Source/WebCore/bindings/js/JSDOMWrapper.h
@@ -20,19 +20,21 @@
  */
 
 #pragma once
+#include "root.h"
+#include "ZigGlobalObject.h"
 
+#include "JSDOMGlobalObject.h"
+#include "NodeConstants.h"
 #include <JavaScriptCore/JSDestructibleObject.h>
-#include <JavaScriptCore/JSEmbedderArrayLike.h>
-#include <WebCore/JSDOMGlobalObject.h>
-#include <WebCore/NodeType.h>
 #include <wtf/SignedPtr.h>
 
-namespace JSC {
-class Structure;
-}
-
 namespace WebCore {
-
+using namespace Zig;
+#ifndef RENAMED_JSDOM_GLOBAL_OBJECT
+#define RENAMED_JSDOM_GLOBAL_OBJECT
+using JSDOMGlobalObject = Zig::GlobalObject;
+}
+#endif
 class ScriptExecutionContext;
 
 // JSC allows us to extend JSType. If the highest 3 bits are set, we can add any Object types and they are
@@ -43,26 +45,23 @@ class ScriptExecutionContext;
 // offset | 7 | 6 | 5 | 4 | 3   2   1   0  |
 // value  | 1 | 1 | 1 | 1 |    NodeType    |
 
-static constexpr uint8_t JSEmbedderArrayLikeType        = JSC::EmbedderArrayLikeType;
-static constexpr uint8_t JSDOMWrapperType                = 0b11101110;
-static constexpr uint8_t JSEventType                     = 0b11101111;
-static constexpr uint8_t JSNodeType                      = 0b11110000;
-static constexpr uint8_t JSNodeTypeMask                  = 0b00001111;
-static constexpr uint8_t JSTextNodeType                  = JSNodeType | std::to_underlying(NodeType::Text);
-static constexpr uint8_t JSProcessingInstructionNodeType = JSNodeType | std::to_underlying(NodeType::ProcessingInstruction);
-static constexpr uint8_t JSDocumentTypeNodeType          = JSNodeType | std::to_underlying(NodeType::DocumentType);
-static constexpr uint8_t JSDocumentFragmentNodeType      = JSNodeType | std::to_underlying(NodeType::DocumentFragment);
-static constexpr uint8_t JSDocumentWrapperType           = JSNodeType | std::to_underlying(NodeType::Document);
-static constexpr uint8_t JSCommentNodeType               = JSNodeType | std::to_underlying(NodeType::Comment);
-static constexpr uint8_t JSCDATASectionNodeType          = JSNodeType | std::to_underlying(NodeType::CDATASection);
-static constexpr uint8_t JSAttrNodeType                  = JSNodeType | std::to_underlying(NodeType::Attribute);
-static constexpr uint8_t JSElementType                   = 0b11110000 | std::to_underlying(NodeType::Element);
+inline constexpr uint8_t JSDOMWrapperType = 0b11101110;
+inline constexpr uint8_t JSEventType = 0b11101111;
+inline constexpr uint8_t JSNodeType = 0b11110000;
+inline constexpr uint8_t JSNodeTypeMask = 0b00001111;
+inline constexpr uint8_t JSTextNodeType = JSNodeType | NodeConstants::TEXT_NODE;
+inline constexpr uint8_t JSProcessingInstructionNodeType = JSNodeType | NodeConstants::PROCESSING_INSTRUCTION_NODE;
+inline constexpr uint8_t JSDocumentTypeNodeType = JSNodeType | NodeConstants::DOCUMENT_TYPE_NODE;
+inline constexpr uint8_t JSDocumentFragmentNodeType = JSNodeType | NodeConstants::DOCUMENT_FRAGMENT_NODE;
+inline constexpr uint8_t JSDocumentWrapperType = JSNodeType | NodeConstants::DOCUMENT_NODE;
+inline constexpr uint8_t JSCommentNodeType = JSNodeType | NodeConstants::COMMENT_NODE;
+inline constexpr uint8_t JSCDATASectionNodeType = JSNodeType | NodeConstants::CDATA_SECTION_NODE;
+inline constexpr uint8_t JSAttrNodeType = JSNodeType | NodeConstants::ATTRIBUTE_NODE;
+inline constexpr uint8_t JSElementType = 0b11110000 | NodeConstants::ELEMENT_NODE;
+inline constexpr uint8_t JSAsJSONType = JSElementType;
 
 static_assert(JSDOMWrapperType > JSC::LastJSCObjectType, "JSC::JSType offers the highest bit.");
-static_assert(JSEmbedderArrayLikeType > JSC::LastJSCObjectType, "EmbedderArrayLikeType must be above LastJSCObjectType.");
-static_assert(JSEmbedderArrayLikeType != JSDOMWrapperType && JSC::EmbedderArrayLikeType != JSEventType, "EmbedderArrayLikeType must not collide with other WebCore types.");
-static_assert(JSEmbedderArrayLikeType < JSNodeType, "EmbedderArrayLikeType must be below JSNodeType to avoid breaking node range checks.");
-static_assert(lastNodeType <= JSNodeTypeMask, "NodeType should be represented in 4bit.");
+static_assert(NodeConstants::LastNodeType <= JSNodeTypeMask, "NodeType should be represented in 4bit.");
 
 class JSDOMObject : public JSC::JSDestructibleObject {
 public:
@@ -71,12 +70,11 @@ public:
     template<typename, JSC::SubspaceAccess>
     static void subspaceFor(JSC::VM&) { RELEASE_ASSERT_NOT_REACHED(); }
 
-    JSDOMGlobalObject* realm() const { return uncheckedDowncast<JSDOMGlobalObject>(JSC::JSNonFinalObject::realm()); }
-    ScriptExecutionContext* scriptExecutionContext() const { return realm()->scriptExecutionContext(); }
+    JSDOMGlobalObject* globalObject() const { return uncheckedDowncast<JSDOMGlobalObject>(JSC::JSNonFinalObject::globalObject()); }
+    ::WebCore::ScriptExecutionContext* scriptExecutionContext() const { return globalObject()->scriptExecutionContext(); }
 
 protected:
     WEBCORE_EXPORT JSDOMObject(JSC::Structure*, JSC::JSGlobalObject&);
-    WEBCORE_EXPORT void finishCreation(JSC::VM&);
 };
 
 template<typename ImplementationClass, typename PtrTraits = RawPtrTraits<ImplementationClass>>
@@ -85,41 +83,24 @@ public:
     using Base = JSDOMObject;
     using DOMWrapped = ImplementationClass;
 
-    ImplementationClass& wrapped() const LIFETIME_BOUND { return m_wrapped; }
-    static constexpr ptrdiff_t offsetOfWrapped() { return OBJECT_OFFSETOF(JSDOMWrapper, m_wrapped); }
+    ImplementationClass& wrapped() const { return m_wrapped; }
+    Ref<ImplementationClass> protectedWrapped() const { return m_wrapped; }
+    static ptrdiff_t offsetOfWrapped() { return OBJECT_OFFSETOF(JSDOMWrapper, m_wrapped); }
     constexpr static bool hasCustomPtrTraits() { return !std::is_same_v<PtrTraits, RawPtrTraits<ImplementationClass>>; };
-    
+
 protected:
     JSDOMWrapper(JSC::Structure* structure, JSC::JSGlobalObject& globalObject, Ref<ImplementationClass>&& impl)
         : Base(structure, globalObject)
-        , m_wrapped(WTF::move(impl)) { }
+        , m_wrapped(WTF::move(impl))
+    {
+    }
 
 private:
-    const Ref<ImplementationClass, PtrTraits> m_wrapped;
+    Ref<ImplementationClass, PtrTraits> m_wrapped;
 };
 
 template<typename ImplementationClass> struct JSDOMWrapperConverterTraits;
 
 JSC::JSValue cloneAcrossWorlds(JSC::JSGlobalObject&, const JSDOMObject& owner, JSC::JSValue);
 
-template<typename ImplementationClass>
-class JSDOMEmbedderArrayLikeWrapper : public JSC::JSEmbedderArrayLike {
-public:
-    using Base = JSC::JSEmbedderArrayLike;
-    using DOMWrapped = ImplementationClass;
-
-    template<typename, JSC::SubspaceAccess>
-    static void subspaceFor(JSC::VM&) { RELEASE_ASSERT_NOT_REACHED(); }
-
-    JSDOMGlobalObject* realm() const { return uncheckedDowncast<JSDOMGlobalObject>(JSC::JSNonFinalObject::realm()); }
-    ScriptExecutionContext* scriptExecutionContext() const { return realm()->scriptExecutionContext(); }
-
-    ImplementationClass& wrapped() const { return static_cast<ImplementationClass&>(embeddedArrayLike()); }
-    constexpr static bool hasCustomPtrTraits() { return false; };
-
-protected:
-    JSDOMEmbedderArrayLikeWrapper(JSC::Structure* structure, JSC::JSGlobalObject& globalObject, Ref<ImplementationClass>&& impl)
-        : Base(globalObject.vm(), structure, WTF::move(impl)) { }
-};
-
 } // namespace WebCore
