diff --git a/Source/bmalloc/mimalloc/mimalloc/CMakeLists.txt b/Source/bmalloc/mimalloc/mimalloc/CMakeLists.txt
index ee043aeacf..083b011be0 100644
--- a/Source/bmalloc/mimalloc/mimalloc/CMakeLists.txt
+++ b/Source/bmalloc/mimalloc/mimalloc/CMakeLists.txt
@@ -535,7 +535,7 @@ function(find_link_library libname outlibname)
 endfunction()
 
 if(WIN32)
-  list(APPEND mi_libraries psapi shell32 user32 advapi32 bcrypt)
+  list(APPEND mi_libraries psapi shell32 user32 advapi32)
 else()
   find_link_library("pthread" MI_LIB_PTHREAD)
   if(MI_LIB_PTHREAD)
