diff --git a/icu4c/source/data/makedata.mak b/icu4c/source/data/makedata.mak
index 71c5d7f..f653533 100644
--- a/icu4c/source/data/makedata.mak
+++ b/icu4c/source/data/makedata.mak
@@ -263,7 +263,7 @@ COMMON_ICUDATA_ARGUMENTS=$(COMMON_ICUDATA_ARGUMENTS) -a ARM64
 #
 #############################################################################
 !IF "$(ARM_CROSS_BUILD)" == ""
-ALL : GODATA "$(ICU_LIB_TARGET)" "$(TESTDATAOUT)\testdata.dat"
+ALL : GODATA "$(ICU_LIB_TARGET)"
 	@echo All targets are up to date
 
 !IF "$(UWP)" == "UWP"
@@ -272,7 +272,7 @@ ALL : GODATA "$(ICU_LIB_TARGET)" "$(TESTDATAOUT)\testdata.dat"
 !ENDIF
 
 !ELSE
-ALL : GODATA "$(ICU_LIB_TARGET)" "$(TESTDATAOUT)\testdata.dat" $(ARM_CROSSBUILD_TS)
+ALL : GODATA "$(ICU_LIB_TARGET)" $(ARM_CROSSBUILD_TS)
 	@echo All targets are up to date
 
 !ENDIF
